src/schema/feeds.ts

```diff
@@ -2195,19 +2289,17 @@ export const resolvers: IResolvers<unknown, BaseContext> = traceResolvers<
       const feed = await feedRepo.save({
         id: await generateShortId(),
         userId: ctx.userId,
-        flags: {
-          name,
-        },
+        flags,
       });
 
       return feed;
     },
     updateFeed: async (
       _,
-      { feedId, name }: { feedId: string; name: string },
+      { feedId, ...flags }: { feedId: string } & FeedFlagsPublic,
       ctx: AuthContext,
     ): Promise<GQLFeed> => {
-      validateFeedPayload({ name });
+      validateFeedPayload({ name: flags.name });
```

Comment:
Can you add validation of icon inside this function with `isOneValidEmoji`, of course its optional as well.

Reply:
Added